google/cloud/spanner/query_partition.cc

```diff
@@ -20,19 +20,24 @@ namespace cloud {
 namespace spanner {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 
-QueryPartition::QueryPartition(std::string transaction_id,
+// Local extension to google::spanner::v1::ExecuteSqlRequest.
+constexpr int kRouteToLeaderFieldNumber = 511037314;
```

Comment:
Quite the magical number, where does it come from?

Reply:
My understanding of the convention to allocate extension field numbers in Google-owned protos is to use the CL number of the change that introduces them.  So, I consumed a couple of CLs to use here.  [We could have just used one, seeing as they are different message types, but separate numbers seemed clearer.]

I didn't want to commit a description like that to the source code, but would be happy to add anything you think might be appropriate.

Reply:
Optional: maybe just say "reserved using Google's conventions" ?

Reply:
SGTM.  Done.  Thanks.